lib/extensions/ar_slug.rb

```diff
@@ -0,0 +1,13 @@
+module ArSlug
+  extend ActiveSupport::Concern
+
+  included do
+    virtual_column :href_slug, :type => :string
+
+    def href_slug
+      return unless id
+      collection = Api::CollectionConfig.new.name_for_subklass(self.class)
+      "#{collection}/#{id}" if collection
```

Comment:
Does this assume that all objects that are configured in the API have a primary collection?